src/effects/VST/VSTEffect.cpp

```diff
@@ -935,9 +935,10 @@ namespace
       // Make a message from a single parameter
       explicit VSTEffectMessage(int id, double value, size_t numParams)
       {
-         mParamsVec.resize(numParams, std::nullopt);
-         if (id < numParams)
-            mParamsVec[id] = value;
+         if (id >= numParams)
+            wxLogDebug(wxT("VST parameter id out of nominal bounds!"));
+         mParamsVec.resize(std::max<size_t>(id + 1, numParams), std::nullopt);
```

Comment:
Looks like it potentially can allocate an arbitrary amount of memory

Reply:
Yes, why I would rather either consider it a bug in the plugin, or else represent the array of parameters sparsely.  The second option is more work.